### DISCUSSION TITLE
Possibility and best approach to define page names when there are several tables on one page

### DISCUSSION THREAD

**Discussion by lsmedia** [Power-Components/livewire-powergrid#1233] [Q&A] [answered]
Is there a way to assign a 'page name' to the paginator of a table so that pagination works correctly on multiple tables per page?

By default, all tables use the page name 'page', which sometimes leads to unwanted behavior when the whole page is reloaded (due to the ambiguous GET parameter `?page=XYZ`).

In the source code I have seen that all parts (especially the pagination view) work entirely without the additional optional parameter `$pageName`. But is there a clean solution to make this work without big changes to the package?

**Comment by lsmedia** [comment]
Does anyone have any ideas?

**Comment by luanfreitasdev** [comment, marked as answer]
```php
    #[Url(as: 'page', keep: true)]
    public string|int $page = 1;

    public function updatedPaginators($value): void
    {
        $this->page = $value;
    }

    public function hydratePage(): void
    {
        $this->paginators['page'] = $this->page;
    }
```

**Reply by luanfreitasdev** [reply]
https://demo.livewire-powergrid.com/filters?page=2

**Reply by luanfreitasdev** [reply]
I believe this is the best way. If you have several tables on the same page, you could assign the name to the alias:

```php
    #[Url(as: 'myCustomPage')]
    public string|int $page = 1;
```

**Reply by lsmedia** [reply]
Thank you very much for drawing my attention to this elegant solution, @luanfreitasdev.

Your approach is wonderful, but there were two problems in my application environment.

#.1 The default query string `page` was partially added to the URL (in addition) to the alias parameters. As a result, the aliases were ignored and the value of `?page=` was always used for all paginators.

#.2 After a reload of the page, the values of the alias parameters were not transferred to `$this->paginators['page']`.

I was able to solve both by adding the following boot method to your code:

```
public function booted(): void
{
    $this->paginators['page'] = $this->page;
}
```

**Comment by lsmedia** [comment]
[Update]

@luanfreitasdev has created a great enhancement for Powergrid, which solves this problem very elegantly in version 5.x.

`
Footer::make()->pageName('customPageName')`

See the documentation: https://livewire-powergrid.com/table/features-setup.html#custom-pagename-in-url
